src/components/Inputs/Checkbox.tsx

```diff
@@ -1,21 +1,28 @@
-import React, { forwardRef, useState } from 'react';
+import React, { forwardRef, ReactElement, Ref, useState } from 'react';
 import cn from 'classnames';
-import { CheckBoxIcon } from 'components/Icons/CheckBoxIcon';
+import CheckBoxIcon from 'components/Icons/CheckBoxIcon';
 
 interface Props extends React.ComponentPropsWithRef<'input'> {
   label?: string;
 }
 
-export const Checkbox = forwardRef<HTMLInputElement, Props>(({ label, className, id, ...rest }, ref) => {
+function Checkbox({ label, className, id, ...rest }: Props, ref: Ref<HTMLInputElement>): ReactElement {
   const [checked, setChecked] = useState<boolean>(false);
-
   return (
-    <div className={cn('flex w-fit cursor-pointer select-none items-center gap-2', className)} onClick={() => setChecked(!checked)}>
-      <input className="hidden" type="checkbox" checked={checked} id={id} ref={ref} {...rest}></input>
+    <div
+      className={cn('flex w-fit cursor-pointer select-none items-center gap-2', className)}
+      onClick={() => setChecked(!checked)}
+      onKeyDown={() => setChecked(!checked)}
+      role="button"
+      tabIndex={0}
+    >
+      <input className="hidden" type="checkbox" checked={checked} id={id} ref={ref} {...rest} />
       <CheckBoxIcon checked={checked} />
       <span>{label}</span>
     </div>
   );
-});
+}
 
 Checkbox.displayName = 'Checkbox';
```

Comment:
as we're now using named functions, setting the `displayName` may no longer be necessary for forwarded ref components